x-pack/plugins/timelines/common/utils/field_formatters.ts

```diff
@@ -118,7 +118,7 @@ export const getDataFromFieldsHits = (
       ? item
           .reduce((acc, i) => [...acc, getDataFromFieldsHits(i, dotField, fieldCategory)], [])
           .flat()
-      : getDataFromFieldsHits(item, prependField, fieldCategory);
+      : getDataFromFieldsHits(item, dotField, fieldCategory);
```

Comment:
Here's an example of a field that gets returned without my fix (this is broken and doesn't work in the UI):

```
 {
    category: 'kibana',
    field: 'criteria.timeUnit',
    values: [ 'm' ],
    originalValue: [ 'm' ],
    isObjectArray: false
  },
```

And here's an example AFTER my fix
```
{
    category: 'kibana',
    field: 'kibana.alert.rule.parameters.criteria.timeUnit',
    values: [ 'm' ],
    originalValue: [ 'm' ],
    isObjectArray: false
  },
```

cc @XavierM @marshallmain

Reply:
Yes that makes sense to me

Reply:
Above change broke unit tests, so I included a not so elegant fix https://github.com/elastic/kibana/pull/120758/commits/b7d8d204577af1b834f3654a4a748bee400ea4f2#diff-cdd49607e6de50d407a10b887f6bf2c930cadb11663d26649b56438f9727548cR120

Any recommendation how it can be done better? @marshallmain

Reply:
I'm not very familiar with the timeline code, but it seems to me that we may always want to pass `dotField` in to the `getDataFromFieldsHits` call? The line above passes `dotField` in to the next call to `getDataFromFieldHits` and `dotField` is `prependField` concatenated with the current field name, which makes sense as that's a logical prefix for any fields nested under the current field. So maybe we don't need special handling for the parameters field and this is just a bug?

Reply:
I think the unit test is wrong in this case and needs to be fixed - what you had originally looks good to me!

Reply:
ya agree with @marshallmain here, https://github.com/elastic/kibana/blob/main/packages/kbn-securitysolution-t-grid/src/mock/mock_event_details.ts#L75 if below this there was another anObject2: { neat: false } everything would be wrong